testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeTimeTravelCompatibility.java

```diff
@@ -72,4 +72,38 @@ public void testReadFromTableRestoredToPreviousVersion()
             dropDeltaTableWithRetry("default." + tableName);
         }
     }
+
+    @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testSelectForVersionAsOf()
+    {
+        String tableName = "test_dl_select_version" + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                " (id INT, part STRING)" +
+                "USING delta " +
+                "PARTITIONED BY (part)" +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'");
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 'spark')");
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " ADD COLUMN new_column INT");
+
+            // Both Spark and Trino Delta Lake connector use the old table definition for the versioned query
+            List<QueryAssert.Row> expectedRows = ImmutableList.of(row(1, "spark"));
+            assertThat(onDelta().executeQuery("SELECT * FROM default." + tableName + " VERSION AS OF 1"))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName + " FOR VERSION AS OF 1"))
+                    .containsOnly(expectedRows);
+
+            // Do time travel after table replacement
+            onDelta().executeQuery("CREATE OR REPLACE TABLE " + tableName + " USING DELTA AS SELECT id + 1 AS id, part, new_column FROM " + tableName);
```

Comment:
Can you pls add an integration test (not product test) having the scenario:
 - create a table with a set of columns
 - do some inserts on it
 - create or replace the table with a different columns
 - check if time travel works as expected.

Reply:
I don't think we can add integration tests with CREATE OR REPLACE. The connector doesn't support the statement yet. https://github.com/trinodb/trino/pull/19991